modules/platform/nuxeo-platform-query-api/src/main/java/org/nuxeo/ecm/platform/query/core/PageProviderServiceImpl.java

```diff
@@ -44,31 +49,104 @@
  */
 public class PageProviderServiceImpl extends DefaultComponent implements PageProviderService {
 
+    private static final Logger log = LogManager.getLogger(PageProviderServiceImpl.class);
+
     private static final long serialVersionUID = 1L;
 
     public static final String PROVIDER_EP = "providers";
 
     // @since 6.0
     public static final String REPLACER_EP = "replacers";
 
+    // definitions added through API, kept for compatibility with JSF ContentViewService
+    protected Map<String, PageProviderDefinition> programmaticDefinitions;
+
+    protected Map<String, Class<? extends PageProvider<?>>> replacerMap;
+
+    @Override
+    public void activate(ComponentContext context) {
+        programmaticDefinitions = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public void deactivate(ComponentContext context) {
+        programmaticDefinitions = null;
+    }
```

Comment:
Could it be placed after stop method? I prefer to follow the order of runtime steps, not a strong opinion, wdyt?

Reply:
At this point of migrations, clearly not a strong opinion on my end for that kind of refinement :D
I placed it after the activation because it makes it easier to check that it "undoes" what activation does, I can change it if you prefer but I might have followed this pattern in a lot of other cases already.

Reply:
We can keep the current code.